src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -268,6 +269,16 @@ export const statePatchV2: RequestHandler = async (req, res) => {
 
 				updateFns.push(async (resinApiTx) => {
 					if (Object.keys(deviceBody).length > 0) {
+						// truncate for resilient legacy compatible device state patch so that supervisors don't fail
+						// to update b/c of length violation of 255 (SBVR SHORT TEXT type) for ip and mac address.
+						// sbvr-types does not export SHORT TEXT VARCHAR length 255 to import.
+						if (deviceBody.ip_address) {
+							deviceBody.ip_address = deviceBody.ip_address?.slice(0, macIpAddressesTruncationLength);
+						}
+						if (deviceBody.mac_address) {
+							deviceBody.mac_address = deviceBody.mac_address?.slice(0, macIpAddressesTruncationLength);
```

Comment:
Instead of `macIpAddressesTruncationLength`, should slice at the last index of the space character smaller than the truncation length, to prevent partial mac/ip addresses from being stored

Reply:
Updated logic to account for this with https://github.com/balena-io/open-balena-api/commit/dd736ce43d85bc640f9b08a479ed783f0f56ed9d